hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1350,47 +1356,56 @@ long generateDiffReport(
                 "Old and new key name both are null");
           } else if (oldKeyName == null) { // Key Created.
             String key = resolveBucketRelativePath(isFSOBucket,
-                newParentIdPathMap, newKeyName);
-            DiffReportEntry entry =
-                SnapshotDiffReportOzone.getDiffReportEntry(CREATE, key);
-            createDiffs.add(codecRegistry.asRawData(entry));
+                newParentIdPathMap, newKeyName, true);
+            if (key != null) {
+              DiffReportEntry entry =
+                  SnapshotDiffReportOzone.getDiffReportEntry(CREATE, key);
+              createDiffs.add(codecRegistry.asRawData(entry));
+            }
           } else if (newKeyName == null) { // Key Deleted.
             String key = resolveBucketRelativePath(isFSOBucket,
-                oldParentIdPathMap, oldKeyName);
+                oldParentIdPathMap, oldKeyName, false);
             DiffReportEntry entry =
                 SnapshotDiffReportOzone.getDiffReportEntry(DELETE, key);
             deleteDiffs.add(codecRegistry.asRawData(entry));
           } else if (isDirectoryObject &&
               Arrays.equals(oldKeyName, newKeyName)) {
             String key = resolveBucketRelativePath(isFSOBucket,
-                newParentIdPathMap, newKeyName);
-            DiffReportEntry entry =
-                SnapshotDiffReportOzone.getDiffReportEntry(MODIFY, key);
-            modifyDiffs.add(codecRegistry.asRawData(entry));
+                newParentIdPathMap, newKeyName, true);
+            if (key != null) {
```

Comment:
Is it possible in case of Modify?

Reply:
yeah we can modify and delete the parent directory all together